### ISSUE TITLE
option to silence a warning

### ISSUE THREAD

**Issue by arnaudon** [BlueBrain/NeuroTS#34]
In production, we get a lot of this one https://github.com/BlueBrain/NeuroTS/blob/9627069be8fe828b3ab8dde8d0250935d038098d/neurots/generate/algorithms/tmdgrower.py#L59

can we find a way to silence it to avoid too much log pollution?

**Comment by eleftherioszisis** [comment]
You could control the logging level messages from the production application, no?

It is important for this warning to exist so that unwanted behavior is caught early.

**Comment by arnaudon** [comment]
yes, but warning level is pretty high, but maybe I can silent it for neurots only. I'll look at why we have this, and how to fix prior to production, it sounds better to do indeed!

**Comment by adrien-berchet** [comment]
If you want to remove all these warnings you could do something like this in your application (which application btw?):
```python
tmdgrower_logger = logging.getLogger("neurots.generate.algorithms.tmdgrower")
tmdgrower_logger.disabled = True
```
Or you can increase its logging level with
```python
tmdgrower_logger.setLevel(logging.ERROR)
```

**Comment by arnaudon** [comment]
ah thanks, I'll add that to a current PR, it's region-grower that takes the heat with that! Also I remember benoit added something in morphio, which after a certain number of same warning says: we'll stop showing them, but there are more. Maybe it's too much to implement here.

**Comment by eleftherioszisis** [comment]
By the way, reducing the segment length can fix that warning. Having big segment lengths results in big steps of path distances that "jump" some branching points.

**Comment by arnaudon** [comment]
would this work? https://github.com/BlueBrain/NeuroTS/pull/36

**Comment by arnaudon** [comment]
solved in #43
